plugin/pkg/shoot/validator/admission.go

```diff
@@ -1296,7 +1321,7 @@ func isUnavailableInAtleastOneZone(regions []core.Region, region string, zones [
 	return false
 }
 
-func validateKubeletConfig(fldPath *field.Path, machineTypes []core.MachineType, workerMachineType string, kubeletConfig *core.KubeletConfig) field.ErrorList {
+func validateKubeletConfig(fldPath *field.Path, machineTypes []gardencorev1beta1.MachineType, workerMachineType string, kubeletConfig *core.KubeletConfig) field.ErrorList {
```

Comment:
Why to change all these functions? You have the core cloudprofile already, Can we not use it everywhere?